components/tikv_alloc/src/lib.rs

```diff
@@ -102,3 +105,15 @@ pub use crate::imp::*;
 
 #[global_allocator]
 static ALLOC: imp::Allocator = imp::allocator();
+
+pub use imp::AllocatorError;
```

Comment:
So for different impls our `imp::AllocatorError` are different types?

Reply:
Does it work for `feature = "tcmalloc"`?

Reply:
Additionally, I guess this PR will conflict with contributor's PR: https://github.com/tikv/tikv/pull/4600/files#diff-1448fd2fb9d71d970c453030f1bf76d2

Reply:
> So for different impls our `imp::AllocatorError` are different types?

Yes. It is a leaky abstraction. I wrap the allocator-dependent error type in a single allocator-independent error type to try to avoid anybody noticing. The likelihood that any code will ever actually care about these error values is low, so I could also just throw away the concrete error type if you prefer and return a generic `struct Error;`.

> Does it work for `feature = "tcmalloc"`?

Yes. The tcmalloc module reexports `io:Error` from default.rs. I have re-tested this to be sure.

> Additionally, I guess this PR will conflict with contributor's PR: https://github.com/tikv/tikv/pull/4600/files#diff-1448fd2fb9d71d970c453030f1bf76d2

Yes, and this one as well: https://github.com/tikv/tikv/pull/4524. Those can land first and I'll deal with the rework.

I was hoping this would land for the 3.0 release as it does contain an update to the underlying jemalloc source, but at this late stage it's probably prudent not to do so.